app/src/main/java/ch/epfl/polybazaar/message/ChatMessageDatabase.java

```diff
@@ -0,0 +1,35 @@
+package ch.epfl.polybazaar.message;
+
+import com.google.android.gms.tasks.Task;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import ch.epfl.polybazaar.database.ModelTransaction;
+
+public class ChatMessageDatabase {
+
+    /**
+     * Fetches the requested message
+     * @param id id of the message
+     * @return successful task containing the model instance if it exists, null otherwise. The task
+     * fails if the database is unreachable
+     */
+    public static Task<ChatMessage> fetch(String id) {
+        return ModelTransaction.fetch(ChatMessage.COLLECTION, id, ChatMessage.class);
+    }
+
+    /**
+     * Fetches all the messages belonging to a conversation between two users for a particular listing
+     * @param userEmail1 : the email of the first user
+     * @param userEmail2 : the email of the second user
+     * @param listingID : the ID of the listing the conversation is about
+     * @return successful task containing the model instance if it exists, null otherwise. The task
+     *         fails if the database is unreachable
+     */
+
+    public static Task<List<ChatMessage>> fetchMessagesOfConversation(String userEmail1, String userEmail2, String listingID) {
+        return ModelTransaction.fetchAllWithMultipleFieldsEquality(ChatMessage.COLLECTION, Arrays.asList("sender", "receiver", "listingID"), Arrays.asList(userEmail1, userEmail2, listingID), ChatMessage.class);
+    }
+}
```

Comment:
Is there any reason why this should not be part of the model directly ?

Reply:
Yes, I designed it that way because all models have their related database class. Though I agree this could be the object of a codebase refactoring

Reply:
Yes, but this was because of the old database API. I suggest that we move to a single class schema for models. But it's my fault, I still haven't found the time to clean up all this, it is kind of a mess.